ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java

```diff
@@ -306,9 +306,13 @@ public ReferenceCountedObject<LogEntryProto> retainLog(long index) throws RaftLo
     }
     final ReferenceCountedObject<LogEntryProto> entry = segment.getEntryFromCache(record.getTermIndex());
     if (entry != null) {
-      getRaftLogMetrics().onRaftLogCacheHit();
-      entry.retain();
-      return entry;
+      try {
+        entry.retain();
+        getRaftLogMetrics().onRaftLogCacheHit();
+        return entry;
+      } catch (IllegalStateException ignore) {
+        // the entry could be removed from the cache and released.
+      }
```

Comment:
why ignore this exception?

Reply:
If the entry is already released, it is the same as a cache miss.

Reply:
Just have updated the comment for clarifying it.